custom_components/tahoma/cover.py

```diff
@@ -95,38 +99,31 @@ def update(self):
 
         self.controller.get_states([self.tahoma_device])
 
-        # Set current position.
+        self.update_postion()
+        self.update_tilt_position()
+        self.update_lock()
+
+    def update_position(self):
+
         # Home Assistant: 0 is closed, 100 is fully open.
         # core:ClosureState: 100 is closed, 0 is fully open.
 
+        states = self.tahoma_device.active_states
+
         # Set position for vertical covers
-        if CORE_CLOSURE_STATE in self.tahoma_device.active_states:
-            self._position = 100 - self.tahoma_device.active_states.get(
-                CORE_CLOSURE_STATE
-            )
+        if CORE_CLOSURE_STATE in states:
+            self._position = 100 - states.get(CORE_CLOSURE_STATE)
 
         # Set position for horizontal covers
-        if CORE_DEPLOYMENT_STATE in self.tahoma_device.active_states:
-            self._position = 100 - self.tahoma_device.active_states.get(
-                CORE_DEPLOYMENT_STATE
-            )
+        if CORE_DEPLOYMENT_STATE in states:
+            self._position = 100 - states.get(CORE_DEPLOYMENT_STATE)
 
         # Set position for gates
-        if CORE_PEDESTRIAN_POSITION_STATE in self.tahoma_device.active_states:
-            self._position = 100 - self.tahoma_device.active_states.get(
-                CORE_PEDESTRIAN_POSITION_STATE
-            )
+        if CORE_PEDESTRIAN_POSITION_STATE in states:
+            self._position = 100 - states.get(CORE_PEDESTRIAN_POSITION_STATE)
 
-        if CORE_TARGET_CLOSURE_STATE in self.tahoma_device.active_states:
-            self._position = 100 - self.tahoma_device.active_states.get(
-                CORE_TARGET_CLOSURE_STATE
-            )
-
-        # Set tilt position for slats
-        if CORE_SLATS_ORIENTATION_STATE in self.tahoma_device.active_states:
-            self._tilt_position = 100 - self.tahoma_device.active_states.get(
-                CORE_SLATS_ORIENTATION_STATE
-            )
+        if CORE_TARGET_CLOSURE_STATE in states:
+            self._position = 100 - states.get(CORE_TARGET_CLOSURE_STATE)
 
         if getattr(self, "_position", False):
```

Comment:
```suggestion
        if self._position == False:
```

Could this be simplified like the others? Or would it return `False` if it is None as well?

Reply:
I missed this one you are right. self._position can be only an integer or be None. The ` getattr(self, "_position", False)` was a trick (ugly IMO). 
Before this PR self._position was not always existing. Now I init it in the __init__ method. 
The `getattr(self, "_position", False)` was here to return False in the case where self._position was not existing. We don't need it anymore.